.travis.yml

```diff
@@ -2,43 +2,54 @@ language: go
 
 dist: jammy
 
-go:
-- '1.20.x'
-- '1.21.x'
-- '1.22.x'
-
 notifications:
   email: false
 
 env:
   global:
     - GO111MODULE=on
 
-before_install:
-  - sudo apt-get update
+stages:
+  - name: Build-Test
+  - name: Semantic-Release
+    if: (branch = main) AND (type IN (push, api)) AND (fork = false)
 
+# Default "install" and "script" steps.
 install:
   - curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.55.2
   - curl -sfL https://raw.githubusercontent.com/securego/gosec/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
   - go install golang.org/x/tools/cmd/goimports@latest
-
 script:
   - make tidy
   - make test
   - make lint
   - make scan-gosec
 
-before_deploy:
-  - nvm install 18
-  - node --version
-  - npm --version
-  - npm install
-  - pip install --user bump-my-version
-
-deploy:
-  - provider: script
-    script: npm run semantic-release
-    skip_cleanup: true
-    on:
+jobs:
+  include:
+    - stage: Build-Test
+      language: go
       go: '1.20.x'
-      branch: main
+    - go: '1.21.x'
+    - go: '1.22.x'
+
+    - name: Detect-Secrets
+      language: python
+      python: 3.12
+      install:
+        - pip install --upgrade "git+https://github.com/ibm/detect-secrets.git@master#egg=detect-secrets"
+      script:
+        - detect-secrets scan --update .secrets.baseline
+        - detect-secrets -v audit --report --fail-on-unaudited --fail-on-live --fail-on-audited-real .secrets.baseline
```

Comment:
Just out of curiosity, why not just add these steps to the above "script" section, where the tests are run, etc. ?

This looks good and looks clean overall, I'm just curious why you took this approach.

Reply:
Mainly so that everything is run in parallel.  Under the "Build-Test" stage, there are four different jobs that each run in parallel:
![image](https://github.com/user-attachments/assets/4e23a4c1-21ab-4e4b-b662-b11acfc5b907)

Reply:
And to carry this a little bit further (farther? :joy:), when we merge the PR into main, we'll see the "Semantic-Release" stage get run also:
![image](https://github.com/user-attachments/assets/61cdce6a-cf71-4ef2-b8b3-6632151b4140)

Reply:
Ah okay, cool!